### PR TITLE
Allow easy Docker testing on Mac OS

### PR DESCRIPTION
**Pull request by sanmai**
Testing if flock exists or not needs somewhat more complicated logic than one wants to add to single line in a Makefile. Therefore this wrapper. Should fix the problem from #368 

This PR also includes two small fixes to make Docker testing run more smoothly in most circumstances.

### CODE REVIEW

**Comment by sanmai** [comment]
travis-ci build is failing because of a new rule `native_constant_invocation`, which is going to be rectified by  #371

File: tests/Fixtures/e2e/Parameters_Coverage/composer.json [hunk 1]
```diff
@@ -11,5 +11,11 @@
         "psr-4": {
             "ParamCoverage\\Test\\": "tests/"
         }
+    },
+    "config": {
```
Comment (BackEndTea) [inline tests/Fixtures/e2e/Parameters_Coverage/composer.json]: Is this related to the mac os builds? or is it something else?

Comment (sanmai) [reply tests/Fixtures/e2e/Parameters_Coverage/composer.json]: Nah, that's just a random bug not worthy a separate PR: test dependencies are installed just once, and without an explicit platform definition if the first install happens not from PHP 7.0, all future tests on earlier versions will go broke.

**Comment by borNfreee** [comment]
thank you @sanmai
